packages/ckeditor5-utils/src/collection.ts

```diff
@@ -25,7 +25,17 @@ import mix from './mix';
  *
  * @mixes module:utils/emittermixin~EmitterMixin
  */
-export default class Collection {
+class Collection<T extends { [ id in I ]?: string }, I extends string = 'id'> implements Iterable<T> {
+	private readonly _items: T[];
+	private readonly _itemMap: Map<string, T>;
+	private readonly _idProperty: I;
+	private _bindToCollection?: Collection<any, any> | null;
+	private readonly _bindToExternalToInternalMap: WeakMap<any, T>;
+	private readonly _bindToInternalToExternalMap: WeakMap<T, any>;
+	private _skippedIndexesFromExternal: number[];
```

Comment:
JSDocs shouldn't be here? As discussed, it's more readable than when it's in the constructor.